pkg/workloads/manager.go

```diff
@@ -79,24 +79,59 @@ const (
 // characters, hyphens, underscores, and dots.
 var workloadNamePattern = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
 
-// NewManager creates a new container manager instance.
+// noopStatusManager is a no-op implementation of StatusManager for fallback cases
+type noopStatusManager struct{}
+
+func (*noopStatusManager) CreateWorkloadStatus(_ context.Context, _ string) error { return nil }
+func (*noopStatusManager) GetWorkloadStatus(_ context.Context, _ string) (*WorkloadStatus, string, error) {
+	return nil, "", nil
+}
+func (*noopStatusManager) SetWorkloadStatus(_ context.Context, _ string, _ WorkloadStatus, _ string) {
+}
+func (*noopStatusManager) DeleteWorkloadStatus(_ context.Context, _ string) error { return nil }
+
+// NewManager creates a new workload manager
 func NewManager(ctx context.Context) (Manager, error) {
+	// Create a manager for config access
+	manager, err := client.NewManager(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client manager: %w", err)
+	}
+
 	runtime, err := ct.NewFactory().Create(ctx)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create runtime: %w", err)
 	}
 
+	statuses := NewStatusManagerFromRuntime(runtime)
+
 	return &defaultManager{
 		runtime:  runtime,
-		statuses: NewStatusManagerFromRuntime(runtime),
+		statuses: statuses,
+		manager:  manager,
 	}, nil
 }
 
-// NewManagerFromRuntime creates a new container manager instance from an existing runtime.
+// NewManagerFromRuntime creates a new workload manager from an existing runtime
 func NewManagerFromRuntime(runtime rt.Runtime) Manager {
+	// Create a manager for config access
+	manager, err := client.NewManager(context.Background())
+	if err != nil {
+		// If we can't create a manager, we'll use a fallback approach
```

Comment:
It's not clear what the purpose of this fallback is. If the client cannot be created - we should error out. The no-op manager will prevent toolhive from doing any useful work.